provisioning/device/src/Transports/Amqp/AmqpLinkFactory.cs

```diff
@@ -11,61 +11,24 @@ internal class AmqpLinkFactory : ILinkFactory
     {
         public IAsyncResult BeginOpenLink(AmqpLink link, TimeSpan timeout, AsyncCallback callback, object state)
         {
-            return TaskHelpers.ToAsyncResult(OpenLinkAsync(link, timeout), callback, state);
+            Debug.Fail($"{nameof(AmqpLinkFactory)} open link should not be used.");
+            throw new NotImplementedException();
         }
 
         public AmqpLink CreateLink(AmqpSession session, AmqpLinkSettings settings)
         {
-            AmqpLink link;
```

Comment:
Matching impl in hub device layer. This code path never gets hit. Also for EndLink.